sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,6 +21,171 @@
+        >>> domainMatrix.__eq__(domainMatrix1)
+        False
+
+        @rref: Returns reduced-row echelon form and list of pivots for the DomainMatrix
+        >>> dMF = domainMatrix.convert_to(RR)
+        >>> dMF
+        DomainMatrix([[99.0, 16.0, 30.0, 62.0], [65.0, 79.0, 94.0, 15.0], [38.0, 13.0, 31.0, 47.0], [17.0, 2.0, 53.0, 78.0]], (4, 4), RR)
+        >>> dMF.rref()
+        (DomainMatrix([[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]], (4, 4), RR),
+        (0, 1, 2, 3))
+
+        @nullspace: Returns the Null Space for the DomainMatrix
+        >>> dMF.nullspace()
+        DomainMatrix([], (0, 4), RR)
+
+        @lu: Returns Lower and Upper decomposition(L, U triangular matrices) of the DomainMatrix
+        >>> dMF.lu()
+        (DomainMatrix([[1.0, 0.0, 0.0, 0.0], [0.656565656565657, 1.0, 0.0, 0.0], [0.383838383838384, 0.100132723787052, 1.0, 0.0], [0.171717171717172, -0.0109128447131691, 4.03990205081114, 1.0]], (4, 4), RR),
+        DomainMatrix([[99.0, 16.0, 30.0, 62.0], [0.0, 68.4949494949495, 74.3030303030303, -25.7070707070707], [0.0, 0.0, 12.0446836749742, 25.7761392125055], [0.0, 0.0, 0.0, -37.060079583716]], (4, 4), RR), [])
+
+
+        @lu_solve: Solver for "DomainMatrix" x in the A*x = B, where A, B are DomainMatrix matrices.
+        >>> A = Matrix([[77, 97], [76, 31]])
+        >>> B = Matrix([[54, 30], [72, 77]])
+        >>> A1 = DomainMatrix.from_Matrix(A)
+        >>> B1 = DomainMatrix.from_Matrix(B)
+        >>> A1
+        DomainMatrix([[77, 97], [76, 31]], (2, 2), ZZ)
+        >>> B1
+        DomainMatrix([[54, 30], [72, 77]], (2, 2), ZZ)
+        >>> A1f = A1.convert_to(RR)
+        >>> B1f = B1.convert_to(RR)
+        >>> A1f.lu_solve(B1f)
+        DomainMatrix([[1.06519558676028, 1.31173520561685], [-0.288866599799398, -0.731995987963892]], (2, 2), RR)
+
+
+        >>> Matrix1 = Matrix([[2, 73], [27, 86]])
+        >>> domainMatrix = DomainMatrix.from_Matrix(Matrix1)
+        >>> Matrix2 = Matrix([[19, 30], [56, 14]])
+        >>> domainMatrix1 = DomainMatrix.from_Matrix(Matrix2)
+        >>> domainMatrix
+        DomainMatrix([[2, 73], [27, 86]], (2, 2), ZZ)
+        >>> domainMatrix1
+        DomainMatrix([[19, 30], [56, 14]], (2, 2), ZZ)
+
+        @mul: Performs term by term multiplication for the second DomainMatrix w.r.t first DomainMatrix.
+        Returns a DomainMatrix whose rows are list of DomainMatrix matrices created after term by term multiplication.
+        >>> domainMatrix.mul(domainMatrix1)
+        DomainMatrix([[DomainMatrix([[38, 60], [112, 28]], (2, 2), ZZ), DomainMatrix([[1387, 2190], [4088, 1022]], (2, 2), ZZ)],
+        [DomainMatrix([[513, 810], [1512, 378]], (2, 2), ZZ), DomainMatrix([[1634, 2580], [4816, 1204]], (2, 2), ZZ)]], (2, 2), ZZ)
+
+        @matmul: Performs matrix multiplication of two DomainMatrix matrices
+        >>> domainMatrix.matmul(domainMatrix1)
+        DomainMatrix([[4126, 1082], [5329, 2014]], (2, 2), ZZ)
```

Comment:
It would be better to add docs for specific methods in docstrings for those methods.

Reply:
The issue with that is that every time we have to define the matrices for showing the usage. We can add the explanation to the methods directly though.
Kindly share your opinion. 
Thanks.

Reply:
I think it's good to show the full steps needed in each docstring. That's what already happens throughout the sympy docs.